huaweicloud/services/obs/data_source_huaweicloud_obs_buckets.go

```diff
@@ -0,0 +1,137 @@
+package obs
+
+import (
+	"context"
+	"errors"
+
+	"github.com/chnsz/golangsdk"
+	"github.com/chnsz/golangsdk/openstack/obs"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/helper/hashcode"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
+)
+
+func DataSourceObsBuckets() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceObsBucketsRead,
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"bucket": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"enterprise_project_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"buckets": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"region": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"bucket": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"enterprise_project_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"storage_class": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"created_at": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceObsBucketsRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	client, err := config.ObjectStorageClient(config.GetRegion(d))
+	if err != nil {
+		return fmtp.DiagErrorf("Error creating HuaweiCloud OBS client: %s", err)
+	}
+	r, err := client.ListBuckets(&obs.ListBucketsInput{
+		BucketType: obs.OBJECT,
+	})
+
+	if err != nil {
+		return fmtp.DiagErrorf("Error querying OBS buckets: %s", err)
+	}
+
+	if len(r.Buckets) == 0 {
+		return fmtp.DiagErrorf("Your query returned no results. " +
+			"Please change your search criteria and try again.")
+	}
+	ids := make([]string, 0, len(r.Buckets))
+	buckets := make([]map[string]interface{}, 0, len(r.Buckets))
+	for _, v := range r.Buckets {
+		metadata, err := queryMetadata(client, v.Name)
+		if err != nil && !errors.As(err, &golangsdk.ErrDefault404{}) {
+			return fmtp.DiagErrorf("Error querying OBS bucket metadata: %s", err)
+		}
+
+		storageClass, enterpriseProjectID, region := "", "", ""
+		if metadata != nil {
+			storageClass = string(metadata.StorageClass)
+			enterpriseProjectID = metadata.Epid
+			region = metadata.Location
+		}
+
+		if bucket, ok := d.GetOk("bucket"); ok {
+			if v.Name != bucket {
+				continue
+			}
+		}
+		if epid, ok := d.GetOk("enterprise_project_id"); ok {
```

Comment:
suggest use GetEnterpriseProjectID()

Reply:
this is a data source, so `enterprise_project_id` should take the value from the arguments